### PR TITLE
Completed the Final Report

### PR DESCRIPTION
**Pull request by michplunkett**
## Describe your changes
Finalize the contents of the final report.

## Checklist before requesting a review
- [x] The code runs successfully.

```
Here is some console output
```
- [x] I ran `make lint` and have made the bulk of changes that it has requested.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -47,6 +47,7 @@ There are two ways that you can run this application, one of them is to run all
 - `make logistic-regression`: Runs the [Logistic Regression](https://en.wikipedia.org/wiki/Logistic_regression) model on the tokenized [Convokit](https://convokit.cornell.edu/documentation/supreme.html) data.
 - `make random-forest`: Runs the [Random Forest](https://en.wikipedia.org/wiki/Random_forest) model on the tokenized [Convokit](https://convokit.cornell.edu/documentation/supreme.html) data.
 - `make xg-boost`: Runs the [XGBoost](https://en.wikipedia.org/wiki/XGBoost) model on the tokenized [Convokit](https://convokit.cornell.edu/documentation/supreme.html) data.
+- `make simluate`: Runs a simulation utilizing all three of the above models to predict the verdict for each judge and then uses the majority decision to predict the outcome of each case.
```
Comment (michplunkett) [inline README.md:50]: Added README documentation for `make simulate`.

File: supreme_court_predictions/models/simulation.py [hunk 2]
```diff
@@ -40,39 +45,37 @@ def __init__(
         ]
         data_tuple = self.merge_vectorize_data()
         for model in list_of_models:
-            self.simulate_one_model(input_model=model, data_tuple=data_tuple)
+            self.simulate_model(input_model=model, data_tuple=data_tuple)
 
     def merge_vectorize_data(self):
         local_path = get_full_data_pathway("clean_convokit/")
-        if os.path.isfile(local_path + "utterances_df.p"):
```
Comment (michplunkett) [inline supreme_court_predictions/models/simulation.py:47]: Files are presumed to be there for all commands.

File: supreme_court_predictions/models/simulation.py [hunk 3]
```diff
@@ -40,39 +45,37 @@ def __init__(
...
 
         merged_df = pd.merge(
             simulation_utterance,
             voters,
             left_on=["case_id", "speaker"],
             right_on=["case_id", "voter"],
         )
-        vectorizer = CountVectorizer(analyzer="word", max_features=5000)
+        vectorizer = CountVectorizer(
+            analyzer="word", max_features=self.max_features
```
Comment (michplunkett) [inline supreme_court_predictions/models/simulation.py:72]: Using `self` value that is in the constructor.

File: supreme_court_predictions/models/simulation.py [hunk 4]
```diff
@@ -110,22 +113,23 @@ def simulate_one_model(self, input_model, data_tuple):
...
 
                 except ValueError:
-                    print("Dataset too small for splitting")
+                    debug_print(
+                        "Dataset too small for splitting", self.debug_mode
+                    )
                     continue
 
-                if len(y_test) != 0:
+                if len(y_test):
```
Comment (michplunkett) [inline supreme_court_predictions/models/simulation.py:121]: These are functionally equivalent statements.

File: supreme_court_predictions/models/simulation.py [hunk 5]
```diff
@@ -148,25 +152,24 @@ def simulate_one_model(self, input_model, data_tuple):
...
-            only_preds = [
-                tup[0] for tup in pred_speaker_tuples
-            ]  # Extract all predictions for this case_id
-            counter = Counter(only_preds)
+            # Extract all predictions for this case_id
+            only_predictions = [tup[0] for tup in pred_speaker_tuples]
+            counter = Counter(only_predictions)
 
-            if len(counter.most_common(1)) != 0:
+            if len(counter.most_common(1)):
```
Comment (michplunkett) [inline supreme_court_predictions/models/simulation.py:169]: These are functionally equivalent statements.

File: supreme_court_predictions/models/logistic_regression.py [hunk 6]
```diff
@@ -55,16 +52,14 @@ def __init__(
...
-                # Add name of file
-                self.dataframe_names.append(df.split(".")[0])
+            # Use the correct file reading function
+            read_func = (
+                pd.read_pickle if df.split(".")[-1] == "p" else pd.read_csv
+            )
+            self.dataframes.append(read_func(self.local_path + df))
+
+            # Add name of file
+            self.dataframe_names.append(df.split(".")[0])
```
Comment (michplunkett) [inline supreme_court_predictions/models/logistic_regression.py:62]: Throughout the rest of the application, it is presume files are there. We should let the models error like we do the rest of the app.

**Comment by JJModern** [comment]
Looks great!

**Review by JJModern — APPROVED** [review]

**Pull request merged**